src/ethereum_spec_tools/lint.py

```diff
@@ -92,14 +94,26 @@ def visit_AsyncFunctionDef(self, function: ast.AsyncFunctionDef) -> None:
         Visit an asynchronous function.
         """
         self._insert(function.name)
-        # Explicitly don't visit the children of functions.
+
+        # Detect imports inside function body
+        for node in function.body:
+            if isinstance(node, ast.ImportFrom):
+                self.item_imports.append(str(node.module))
+            if isinstance(node, ast.Import):
+                self.item_imports.append(node.names[0].name)
 
     def visit_FunctionDef(self, function: ast.FunctionDef) -> None:
         """
         Visit a function.
         """
         self._insert(function.name)
-        # Explicitly don't visit the children of functions.
+
+        # Detect imports inside function body
+        for node in function.body:
+            if isinstance(node, ast.ImportFrom):
+                self.item_imports.append(str(node.module))
+            if isinstance(node, ast.Import):
+                self.item_imports.append(node.names[0].name)
```

Comment:
I think you'll need to actually visit the children of functions, because I don't think the current implementation would catch:

```python
def foo():
    if True:
        import ethereum.some_fork.spec
```

Reply:
Have implemented the change. Skipped implementing the `visit_FunctionDef` so it visits the child nodes